src/dataset/mod.rs

```diff
@@ -134,3 +134,62 @@ mod tests {
         assert_eq!(*m[1][3], 9);
     }
 }
+
+impl<X: Copy + std::fmt::Debug, Y: Copy + std::fmt::Debug> std::fmt::Display for Dataset<X, Y> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        struct Target<Y> {
+            name: String,
+            value: Y,
+        }
+        struct Feature<X> {
+            name: String,
+            value: X,
+        }
+        struct DataPoint<X, Y> {
+            labels: Vec<Target<Y>>,
+            features: Vec<Feature<X>>,
+        }
+        impl<X: Copy + std::fmt::Debug, Y: Copy + std::fmt::Debug> std::fmt::Display for DataPoint<X, Y> {
+            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                write!(
+                    f,
+                    "{} : {}",
+                    self.labels
+                        .iter()
+                        .map(|target| format!("{}:{:?}, ", target.name, target.value))
+                        .collect::<String>(),
+                    self.features
+                        .iter()
+                        .map(|feature| format!("{}:{:?}, ", feature.name, feature.value))
+                        .collect::<String>()
+                )
+            }
+        }
+        let mut datapoints = Vec::new();
+        for sample_index in 0..self.num_samples {
+            let mut features = Vec::new();
+            for feature_index in 0..self.feature_names.len() {
+                features.push(Feature {
+                    name: self.feature_names[feature_index].to_owned(),
+                    value: self.data[sample_index * self.num_features + feature_index],
+                });
+            }
+            let mut targets = Vec::new();
+            for target_index in 0..self.target_names.len() {
```

Comment:
What if a dataset does not has names for all target variables? What if the `self.target_names` list does not cover all target variables?